walletclient.go

```diff
@@ -1,117 +1,106 @@
-// Package walletclient is a Go client for interacting with Spv Wallet.
 package walletclient
 
 import (
-	"github.com/bitcoin-sv/spv-wallet-go-client/transports"
-	"github.com/bitcoinschema/go-bitcoin/v2"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/models"
 	"github.com/libsv/go-bk/bec"
 	"github.com/libsv/go-bk/bip32"
-	"github.com/libsv/go-bk/wif"
-	"github.com/pkg/errors"
 )
 
-// ClientOps are used for client options
-type ClientOps func(c *WalletClient)
-
-// WalletClient is the spv wallet go client representation.
+// WalletClient is the spv wallet Go client representation.
 type WalletClient struct {
-	transports.TransportService
-	accessKey        *bec.PrivateKey
-	accessKeyString  string
-	transport        transports.TransportService
-	transportOptions []transports.ClientOps
-	xPriv            *bip32.ExtendedKey
-	xPrivString      string
-	xPub             *bip32.ExtendedKey
-	xPubString       string
+	signRequest bool
+	server      string
+	httpClient  *http.Client
+	accessKey   *bec.PrivateKey
+	adminXPriv  *bip32.ExtendedKey
+	xPriv       *bip32.ExtendedKey
+	xPub        *bip32.ExtendedKey
 }
 
-// New create a new wallet client
-func New(opts ...ClientOps) (*WalletClient, error) {
-	client := &WalletClient{}
+// NewWalletClientWithXPrivate creates a new WalletClient instance using a private key (xPriv).
+// It configures the client with a specific server URL and a flag indicating whether requests should be signed.
+// - `xPriv`: The extended private key used for cryptographic operations.
+// - `serverURL`: The URL of the server the client will interact with.
+func NewWithXPriv(xPriv, serverURL string) *WalletClient {
```

Comment:
It would be great (and would match js-client options) to give optional possibility to optionally add admin key in all those "constructors"